File: ble-livedata/src/main/java/no/nordicsemi/android/ble/livedata/state/BondState.java
package no.nordicsemi.android.ble.livedata.state;

import androidx.annotation.NonNull;
import androidx.annotation.RestrictTo;

/** @noinspection unused*/
public class BondState {

	public enum State {
		NOT_BONDED,
		BONDING,
		BONDED
	}

	public static final class NotBonded extends BondState {
		@RestrictTo(RestrictTo.Scope.LIBRARY)
		public static final NotBonded INSTANCE = new NotBonded();
		private NotBonded() {
			super(State.NOT_BONDED);
		}
	}

	public static final class Bonding extends BondState {
		@RestrictTo(RestrictTo.Scope.LIBRARY)
		public static final Bonding INSTANCE = new Bonding();
		private Bonding() {
			super(State.BONDING);
		}
	}

	public static final class Bonded extends BondState {
		@RestrictTo(RestrictTo.Scope.LIBRARY)
		public static final Bonded INSTANCE = new Bonded();
		private Bonded() {
			super(State.BONDED);
		}
	}

	protected final BondState.State state;

	private BondState(@NonNull final BondState.State state) {
		this.state = state;
	}

	/**
	 * The bonding state. This can be used in <i>switch</i> in Java.
	 */
	public final BondState.State getState() {
		return state;
	}

	/**
	 * Whether bonding was established.
	 */
	public final boolean isBonded() {
		return state == State.BONDED;
	}
}

File: ble-livedata/src/main/java/no/nordicsemi/android/ble/livedata/state/BondState.kt
package no.nordicsemi.android.ble.livedata.state

@Suppress("unused")
sealed class BondState(val state: State) {
    /** The bonding state. This can be used in <i>switch</i> in Java. */
    enum class State {
        NOT_BONDED, BONDING, BONDED
    }

    /** The device was not connected or is not bonded. */
    object NotBonded: BondState(State.NOT_BONDED)

    /** Bonding has started. */
    object Bonding: BondState(State.BONDING)

    /** The device is bonded. */
    object Bonded: BondState(State.BONDED)

    /** Whether bonding was established. */
    val isBonded: Boolean
        get() = this is Bonded
}